test/Ocelot.UnitTests/DownstreamUrlCreator/DownstreamUrlCreatorMiddlewareTests.cs

```diff
@@ -611,6 +611,43 @@ public void Should_map_when_query_parameters_has_same_names_with_placeholder()
         ThenTheQueryStringIs($"?roleId={roleid}&{everything}");
     }
 
+    [Theory]
+    [Trait("Bug", "2116")]
+    [InlineData("api/debug)")]
+    [InlineData("api/debug(")]
+    public void Should_not_fail_to_handle_parentheses(string urlPath)
+    {
+        // Arrange
+        var withGetMethod = new List<string> { "Get" };
+        var downstreamRoute = new DownstreamRouteBuilder()
+            .WithUpstreamPathTemplate(new UpstreamPathTemplateBuilder()
+                .WithOriginalValue("/routed/api/{path}")
+                .Build())
+            .WithDownstreamPathTemplate("/api/{path}")
+            .WithUpstreamHttpMethod(withGetMethod)
+            .WithDownstreamScheme(Uri.UriSchemeHttp)
+            .Build();
+        GivenTheDownStreamRouteIs(new DownstreamRouteHolder(
+            new List<PlaceholderNameAndValue>
+            {
+                new("{path}", urlPath),
+            },
+            new RouteBuilder().WithDownstreamRoute(downstreamRoute)
+                .WithUpstreamHttpMethod(withGetMethod)
+                .Build()
+        ));
+        GivenTheDownstreamRequestUriIs($"http://localhost:5000/{urlPath}");
+        GivenTheServiceProviderConfigIs(new ServiceProviderConfigurationBuilder().Build());
+        GivenTheUrlReplacerWillReturn($"routed/{urlPath}");
+
+        // Act
+        WhenICallTheMiddleware();
+
+        // Assert
+        ThenTheDownstreamRequestUriIs($"http://localhost:5000/routed/{urlPath}");
+        Assert.True(_httpContext.Response.StatusCode < 500);
```

Comment:
It's peculiar... A strict assertion is required in this case, for example `HttpStatusCode.OK`

Reply:
An acceptance test is also necessary!

Reply:
Hi Raman, I will create an additional Acceptance test👍

Reply:
Your **complete code** will undoubtedly be a valuable addition. However, please be aware that we are currently focusing on the hot🔥[v23.3 Hotfixes](https://github.com/ThreeMammals/Ocelot/milestone/8) milestone, so we would greatly appreciate your prompt development of tests and submission of commits. Also, note that we follow Pair Programming practices to accelerate delivery, allowing both you and other code reviewers to commit.

Reply:
I created an Acceptance Test that helped me reproduce and validate the bug successfully.